### PR TITLE
Add Binary entity creation mode

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         - name: Download dependencies
           run: |
             git clone -b foxy https://github.com/eProsima/Micro-CDR src/Micro-CDR 
-            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client 
+            git clone -b feature/binary_entity_creation https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
```
Comment (pablogs9) [inline .github/workflows/ci.yml:25]: Before merge

```suggestion
            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client 
```

**Comment by pablogs9** [comment]
Test not passing due to agent does not have binary functionality. Try again when https://github.com/eProsima/Micro-XRCE-DDS-Agent/pull/239 merged

**Comment by pablogs9** [comment]
Marie Kondo

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,8 +21,8 @@ jobs:
 
         - name: Download dependencies
           run: |
-            git clone -b foxy https://github.com/eProsima/Micro-CDR src/Micro-CDR 
-            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client 
+            git clone -b develop https://github.com/eProsima/Micro-CDR src/Micro-CDR
```
Comment (pablogs9) [inline .github/workflows/ci.yml:24]:
```suggestion
            git clone -b foxy https://github.com/eProsima/Micro-CDR src/Micro-CDR 
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -21,8 +21,8 @@ jobs:
 
         - name: Download dependencies
           run: |
-            git clone -b foxy https://github.com/eProsima/Micro-CDR src/Micro-CDR 
-            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client 
+            git clone -b develop https://github.com/eProsima/Micro-CDR src/Micro-CDR 
+            git clone -b develop https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client
```
Comment (pablogs9) [inline .github/workflows/ci.yml:25]:
```suggestion
            git clone -b foxy https://github.com/eProsima/Micro-XRCE-DDS-Client src/Micro-XRCE-DDS-Client 
```

**Review by Acuadros95 — APPROVED** [review]

**Comment by pablogs9** [comment]
Merging on Tuesday 4th May

**Pull request merged**